src/expression.pest

```diff
@@ -0,0 +1,119 @@
+// See: https://github.com/pest-parser/pest/issues/304
+eoi = _{ !ANY }
+
+// You can find the grammar for actions of UpdateItem in the following link.
+// https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html
+set_action = { SOI ~ path ~ "=" ~ value ~ ("," ~ path ~ "=" ~ value)* ~ eoi }
+remove_action = { SOI ~ path ~ ("," ~ path)* ~ eoi }
+
+path = { attr_access ~ ("." ~ attr_access)* }
+attr_access = _{ attr_name ~ list_elem* }
+
+// Attribute name can be quoted by backticks.
+attr_name = _{ "`" ~ quoted_identifier ~ "`" | non_quoted_identifier }
+
+// You can write attribute name for a`b with a``b like MySQL.
+// https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
+quoted_identifier = @{ ("``" | (!"`" ~ ANY))+ }
+
+// Dynein allows non ASCII path identifier without quoting.
+non_quoted_identifier = @{ (!special_character ~ XID_START) ~ (!special_character ~ XID_CONTINUE)* }
+
+// Backtick (`), double quote ("), single quote (') and back slash (\) have a special meaning in dynein.
+// Currently, DynamoDB does not use an asterisk (*), slash (/), percent sign (%) as a special character.
+// However, dynein treats them as special characters for understandability.
+special_character = _{
+  SEPARATOR | "\\" | "`" | "\"" | "'" | "=" | "<" | "<" | ">" | "[" | "]" | "+" | "-" | "*" | "/" | "%" | "."
+}
+
+// DynamoDB does not allow the plus sign for an index of a list.
+list_elem = _{ "[" ~ list_index_number ~ "]" }
+list_index_number = @{ (ASCII_NONZERO_DIGIT+ ~ ASCII_DIGIT*) | "0" }
+
+// DynamoDB allows parentheses to group expressions but does not allow redundancy.
+value = { "(" ~ plus_expression ~ ")" | "(" ~ minus_expression ~ ")" | plus_expression | minus_expression | operand }
+plus_expression = { operand ~ "+" ~ operand }
+minus_expression = { operand ~ "-" ~ operand }
+
+// Note: Unary operators, plus (+), minus (-) are not allowed in DynamoDB.
+//       But, they can be prepended before a number literal.
+operand = { function | literal | path  }
+function = { list_append_function | if_not_exists_function }
+list_append_function = { ^"list_append" ~ "(" ~ list_append_parameter ~ "," ~ list_append_parameter ~ ")" }
+list_append_parameter = { path | list_literal }
+if_not_exists_function = { ^"if_not_exists" ~ "(" ~ path ~ "," ~ value ~ ")" }
+
+// Literals
+literal = _{
+  boolean_literal | null_literal | b_literal | list_literal | map_literal  | string_literal | number_literal |
+  set_literal
+}
+
+// Boolean literals
+boolean_literal = _{ true_literal | false_literal }
+true_literal = @{ ^"true" }
+false_literal = @{ ^"false" }
+
+// Null literal
+null_literal = @{ ^"null" }
+
+// String literals
+string_literal = _{ double_quote_literal | single_quote_literal }
+double_quote_literal = @{ "\"" ~ ("\\0" | "\\r" | "\\n" | "\\t" | "\\\\" | "\\\"" | "\\'" | (!"\"" ~ ANY))* ~ "\"" }
+single_quote_literal = @{ "'" ~ (!"\'" ~ ANY)* ~ "'" }
```

Comment:
All right!

Reply:
I added a description for string literals at commit [644c86b](https://github.com/awslabs/dynein/pull/132/commits/644c86b5ba2731ed1243dde08fd65ba4134b8f2f). Currently, I refrained from adding a description of binary literals because we have not implemented a display of their value yet.